velox/functions/sparksql/RegexFunctions.cpp

```diff
@@ -57,6 +46,40 @@ struct RegexpReplaceFunction {
 
   static constexpr bool is_default_ascii_behavior = true;
 
+  FOLLY_ALWAYS_INLINE void initialize(
+      const std::vector<TypePtr>& inputTypes,
+      const core::QueryConfig& config,
+      const arg_type<Varchar>* str,
+      const arg_type<Varchar>* pattern,
+      const arg_type<Varchar>* replacement) {
+    initialize(inputTypes, config, str, pattern, replacement, nullptr);
+  }
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const std::vector<TypePtr>& /*inputTypes*/,
+      const core::QueryConfig& config,
+      const arg_type<Varchar>* /*string*/,
+      const arg_type<Varchar>* pattern,
+      const arg_type<Varchar>* replacement,
+      const arg_type<int64_t>* /*position*/) {
+    if (pattern) {
+      const auto processedPattern = prepareRegexpReplacePattern(*pattern);
+      re_.emplace(processedPattern, RE2::Quiet);
+      VELOX_USER_CHECK(
+          re_->ok(),
+          "Invalid regular expression {}: {}.",
+          processedPattern,
+          re_->error());
+
+      if (replacement) {
+        // Constant 'replacement' with non-constant 'pattern' needs to be
+        // processed separately for each row.
```

Comment:
nit: perhaps move this comment to a proper place as this branch is for constant pattern and constant replacement.